### PR TITLE
Added IProvideResolvedType

### PR DESCRIPTION
**Pull request by JKamsker**
I think its pretty self-explainatory

### CODE REVIEW

File: src/GraphQL/Types/IProvideResolvedType.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
```
Comment (sungam3r) [inline src/GraphQL/Types/IProvideResolvedType.cs:4]:
```suggestion
```

File: src/GraphQL/Types/Composite/ListGraphType.cs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 namespace GraphQL.Types
 {
-    public class ListGraphType<T> : ListGraphType
+    public class ListGraphType<T> : ListGraphType, IProvideResolvedType
```
Comment (sungam3r) [inline src/GraphQL/Types/Composite/ListGraphType.cs:5]: Why not on `ListGraphType` ?

Comment (sungam3r) [reply src/GraphQL/Types/Composite/ListGraphType.cs:5]: Same question for `NonNullGraphType`.

**Comment by sungam3r** [comment]
Yes, it is self-explainatory but what is your case/need in this interface?

**Comment by JKamsker** [comment]
Cant tell right now exactly but for some reason i needed to get the root type.

This comment will be updated as ive researched the root cause for my need. @sung3r

**Comment by JKamsker** [comment]
The reason i needed this, is because i want all fieldtypes from an IGraphType:

```csharp
private IEnumerable<FieldType> GetFieldTypes(IGraphType returnType = null)
{
	if (returnType == null)
	{
		returnType = this.source.ReturnType;
	}

	if (returnType is IComplexGraphType cgt)
	{
		return cgt.Fields;
	}

	if ((returnType is ListGraphType || returnType is NonNullGraphType) && returnType is IProvideResolvedType gt)
	{
		return this.GetFieldTypes(gt.ResolvedType);
	}

	throw new NotImplementedException($"Cannot resolve returntype of '{returnType.GetType().FullName}'");
}
```

File: src/GraphQL/Types/Composite/ListGraphType.cs (hunk 2 above)
Comment (sungam3r) [reply src/GraphQL/Types/Composite/ListGraphType.cs]: @JKamsker  ?

**Comment by JKamsker** [comment]
https://github.com/graphql-dotnet/graphql-dotnet/pull/1680/files#diff-ca5a56c1707ab2a0a55aaf8404e2b59eR1957

File: src/GraphQL/Types/Composite/ListGraphType.cs (hunk 2 above)
Comment (sungam3r) [reply src/GraphQL/Types/Composite/ListGraphType.cs]: I mean to inherit `ListGraphType` and `NonNullGraphType` from `IProvideResolvedType`, not `ListGraphType<T>` and `NonNullGraphType<T>>`.

**Comment by sungam3r** [comment]
@JKamsker I see that you change PR but for some reason github shows it in the past. It seems to me that I have not encountered this before. All force pushes always displayed.

**Comment by sungam3r** [comment]
AppVeyor build also shows 2 monts ago date. No matter, merging.

**Pull request merged**